### PR TITLE
Create apt-get release pipeline

### PR DESCRIPTION
**Pull request by kyle-rader**
Creating a new workflow to be triggered off full release to download the signed .deb asset and deploy it.

The workflow does not currently check if the .deb is actually signed, this is something the repo api _will do in the future_ similar to how it currently does for rpm packages.

Builds now failing because of https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/ `set-env` usage in the GitVersioning tools :(

### CODE REVIEW

File: .github/workflows/build-signed-deb.yml
Comment (kyle-rader) [inline .github/workflows/build-signed-deb.yml:53]: This is for https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

**Comment by mjcheetham** [comment]
I've created a PR into the main branch that updates our Nerdbank.GitVersioning to one that uses environment files in GitHub Actions: https://github.com/microsoft/Git-Credential-Manager-Core/pull/224

Once this is merged, we can unblock this branch (might need a rebase).

File: .github/fetch_release.py
Comment (mjcheetham) [inline .github/fetch_release.py:15]: I know access to the GitHub API can be throttled when used unauthenticated. Using the `GITHUB_TOKEN` might be useful, but not required for this PR.

**Review by mjcheetham — APPROVED** [review]
Looks good to me! Just a comment about using the API without auth can lead to throttling by IP address - not something we need to address immediately.

**Review by derrickstolee — APPROVED** [review]

**Comment by kyle-rader** [comment]
Rebased.

**Pull request merged**
